app/src/test/java/com/example/habitsmasher/ProgressTrackerTest.java

```diff
@@ -0,0 +1,89 @@
+package com.example.habitsmasher;
+import org.junit.Before;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import java.util.Date;
+import java.util.UUID;
+
+public class ProgressTrackerTest {
+    private ProgressTracker _progressTracker;
+    private Habit _parentHabit;
+    private static final String SAMPLE_DAYS_OF_THE_WEEK = "MO WE FR";
+
+    @Before
+    public void setUp() {
+        _parentHabit = new Habit("Test Habit", "Test Reason", new Date(), SAMPLE_DAYS_OF_THE_WEEK, true, UUID.randomUUID().toString(), new HabitEventList());
+    }
+
+    @Test
+    public void calculateProgress_noViableDays_expect0() {
+        // Create date that is not viable
+        Date notInDays = new Date(2021, 11, 27);
+        _parentHabit.setDate(notInDays);
+
+        // Create progress tracker
+        _progressTracker = new ProgressTracker(_parentHabit);
+        assertEquals(0, (int) _progressTracker.calculateProgressPercentage());
+
+    }
+
+    @Test
+    public void calculateProgress_noEventsOnDays_expect0() {
+        // Set date that is viable, but do not add events
+        Date notInDays = new Date(2021, 11, 26);
```

Comment:
Can we extract this to a constant so that's it doesn't look like a random date in a few years?

- [ ] extract to constant